src/models/VisualStyleModel/impl/VisualStyleImpl.ts

```diff
@@ -469,62 +277,627 @@ export const createVisualStyleFromCx = (cx: Cx2): VisualStyle => {
-        vpValue: VisualPropertyValueType
-      }[],
+
+    if (cyStyle.autorotate === true) {
+      delete cyStyle.autorotate
+      cyStyle['text-rotation'] = 'autorotate'
     }
-    return edgeView
+
+    const cyEdge = {
+      group: 'edges',
+      data: {
+        id: edge.id,
+        source: edge.s,
+        target: edge.t,
+      },
+      style: cyStyle,
+    }
+
+    return cyEdge
   })
+
   const networkView = {
     id: network.id,
-    nodeViews,
-    edgeViews,
+    nodeViews: cyNodeViews,
+    edgeViews: cyEdgeViews,
   }
 
   return networkView
 }
+
+export const createCyJsStyleSheetView = (
+  vs: VisualStyle,
+  network: Network,
+  nodeTable: Table,
+  edgeTable: Table,
+): {
+  defaultStyle: Stylesheet[]
+  cyNodes: ElementDefinition[]
+  cyEdges: ElementDefinition[]
+  nodeBypasses: Record<IdType, Partial<Record<string, VisualPropertyValueType>>>
+  edgeBypasses: Record<IdType, Partial<Record<string, VisualPropertyValueType>>>
+} => {
+  // edge ids are of the form 'e1', 'e2', etc. but our app stores them as '1', '2', etc.
+  const nodeStyle: Record<
+    string,
+    VisualPropertyValueType | ((node: NodeSingular) => VisualPropertyValueType)
+  > = {}
+  const edgeStyle: Record<
+    string,
+    VisualPropertyValueType | ((edge: EdgeSingular) => VisualPropertyValueType)
+  > = {}
+
+  const nodeBypasses: Record<
+    IdType,
+    Partial<Record<string, VisualPropertyValueType>>
+  > = {}
+  const edgeBypasses: Record<
+    IdType,
+    Partial<Record<string, VisualPropertyValueType>>
+  > = {}
+
+  nodeVisualProperties(vs).forEach((vpName: VisualPropertyName) => {
+    const vp = vs[vpName] as VisualProperty<VisualPropertyValueType>
+    const defaultValue = vp.defaultValue
+    const mapping = vp.mapping
+    const bypassMap = vp.bypassMap
+    const cyStyleName = cyJsVisualPropertyConverter[vpName]?.cyJsVPName
+
+    if (cyStyleName != null) {
+      nodeStyle[cyStyleName] = defaultValue
+
+      if (mapping != null) {
+        switch (mapping.type) {
+          case 'passthrough': {
+            const { attribute } = mapping
+            nodeStyle[cyStyleName] = (
+              node: NodeSingular,
+            ): VisualPropertyValueType => {
+              const row = nodeTable.rows.get(node.data('id'))
+              const column = nodeTable.columns.get(attribute)
+              const value =
+                (column?.alias != null
+                  ? row?.[column.alias]
+                  : row?.[attribute]) ?? column?.defaultValue
+
+              if (!Array.isArray(value) && value != null) {
+                return value
+              }
+              throw new Error(
+                `Error applying passthrough mapping function on node ${
+                  node.data('id') as string
+                }.  Value: ${String(
+                  value,
+                )} should not be an array or undefined`,
+              )
+            }
+            break
+          }
+          case 'discrete': {
```

Comment:
Mapping handling code for generating stylesheets are duplicated for nodes and edges and quite long.  Extract these functions to a seperate mappingImpl file.